### PR TITLE
Add support for code splitting and static assets

### LINKED ISSUE #61: [React] Support static CRA assets
The Create React App used by this archetype will create JS, CSS and static assets under the build folder:

image

However, aem-clientlib-generator used by this library will only copy the JS and CSS files.
As a result, when static resources are used by the app (Fonts, background images, etc) - They will be broken in AEM.

### LINKED ISSUE #83: React Code splitting
Based on going through the Getting started project [here](https://github.com/Adobe-Marketing-Cloud/aem-guides-wknd-events), it seems like the approach of using `aem-clientlib-generator` does not allow for utilizing code splitting for react in AEM. I posted an issue about this on that project and it was suggested that i post the question here.

https://github.com/Adobe-Marketing-Cloud/aem-guides-wknd-events/issues/52

### PR DESCRIPTION
**Pull request by samuelmeuli**
| Q                        | A 
| ------------------------ | ---
| Fixed Issues?            | Closes #61, closes #83
| Patch: Bug Fix?          | No
| Minor: New Feature?      | Yes
| Major: Breaking Change?  | No
| Tests Added + Pass?      | No
| Documentation Provided   | Yes
| Any Dependency Changes?  | Yes
| License                  | Apache License, Version 2.0

See the newly added doc files for a description of the changes.

_Requires https://github.com/facebook/create-react-app/pull/7721 and https://github.com/wcm-io-frontend/aem-clientlib-generator/pull/23 to be merged and released first._

### CODE REVIEW

**Comment by grubyak** [comment]
`test-spa-react-react-app: Failed to run task: 'npm run build' failed` -- did you see that?

**Comment by samuelmeuli** [comment]
@grubyak Yeah, it's not working yet because it builds on the code of the currently unreleased PRs. Once they're released, we can update the dependencies and the tests should pass :)

**Comment by samuelmeuli** [comment]
`create-react-app` v3.2 is released and includes our change.

**Comment by samuelmeuli** [comment]
https://github.com/wcm-io-frontend/aem-clientlib-generator/pull/23 has also been merged and released, so this PR can now be reviewed :)

File: docs/code-splitting.md [hunk 1]
```diff
@@ -0,0 +1,21 @@
+# Code Splitting
+
+## Overview
+
+The React app is configured to make use of [code splitting](https://webpack.js.org/guides/code-splitting) by default. When building the app for production, the code will be output in several chunks:
```
Comment (habansal) [inline docs/code-splitting.md]: To me, this documentation here seems to say that the Code Splitting is happening only for React app.
It should be made clear that the Code Splitting is supported for both

Comment (samuelmeuli) [reply docs/code-splitting.md]: I've only set it up for React. For Angular, there is route-level code splitting, which won't work with `@adobe/cq-spa-page-model-manager`. And component-level code splitting looks more complicated to set up, so I'm not sure if this should by in the archetype by default. See https://blog.angularindepth.com/dynamically-loading-components-with-angular-cli-92a3c69bcd28

Comment (samuelmeuli) [reply docs/code-splitting.md]: If a user wants code splitting and has updated their code to support it, they'll only have to change a few lines in `clientlib.config.js` so the library picks up the chunks and includes them in the `js` ClientLib folder.

**Pull request merged**